File: Data Structures & Algorithms/search-2d-matrix/submission-1.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int left;
        int right;
        int numberOfRows = 0;
        int mid;

        left = 0;
        right = matrix[numberOfRows].length - 1;

        while(numberOfRows < matrix.length)
        {
            mid = (left + right) / 2;

            if(matrix[numberOfRows][mid] == target)
            {
                return true;
            }
            //right
            else if(target > matrix[numberOfRows][mid])
            {
                left = mid + 1;
            }  
            //left
            else
            {
                right = mid - 1;
            }

            if(left > right)
            {
                numberOfRows++;
                if(numberOfRows < matrix.length)
                {
                    left = 0;
                    right = matrix[numberOfRows].length - 1;
                }
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/first-unique-character-in-a-string/submission-1.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        int n = s.length();
        vector<int>dc(26, 0);
        for(int i=0; i<n; i++)
            dc[s[i]-'a']++;
        
        for(int i=0; i<n; i++)
        {
            if(dc[s[i]-'a'] == 1)
                return i;
        }
        return -1;
    }
};